src/components/application_manager/src/resumption/resumption_data_json.cc

```diff
@@ -106,16 +110,19 @@ void ResumptionDataJson::SaveApplication(
   json_app[strings::time_stamp] = time_stamp;
   json_app[strings::subscribed_for_way_points] = is_subscribed_for_way_points;
 
+  accessor.GetMutableData().set_dictionary(dictionary);
   LOG4CXX_DEBUG(logger_, "SaveApplication : " << json_app.toStyledString());
 }
 
 bool ResumptionDataJson::IsHMIApplicationIdExist(uint32_t hmi_app_id) const {
   using namespace app_mngr;
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock autolock(resumption_lock_);
-  for (Json::Value::iterator it = GetSavedApplications().begin();
-       it != GetSavedApplications().end();
-       ++it) {
+  resumption::LastStateAccessor accessor = last_state_wrapper_->get_accessor();
+  Json::Value dictionary = accessor.GetData().dictionary();
+  const Json::Value& saved_apps = GetSavedApplications(dictionary);
+
+  for (auto it = saved_apps.begin(); it != saved_apps.end(); ++it) {
```

Comment:
maybe range-based

Reply:
@SNiukalov fixed in https://github.com/smartdevicelink/sdl_core/pull/2626/commits/f4e32226bda388a572a9edc282e74636a66ad925